### PR TITLE
Rewrite base.scala.stg using string interpolation

### PR DESCRIPTION
**Pull request by guersam**
This PR replaces `base.scala.stg` with string interpolation. Generated codes are almost identical to the existing except a few whitespace formats.

Because string interpolation cannot handle indents out-of-box, I've added Scalariform to format generated code.

### CODE REVIEW

**Comment by guersam** [comment]
The rewritten code generator is almost one-to-one translation of the previous `base.scala.stg` currently, so it still has some kind of messy workarounds for StringTemplate, which should be refactored in the future.

File: scaloid-common/src/main/st/org/scaloid/common/preferences.scala [hunk 1]
```diff
@@ -119,7 +121,7 @@ class Extra(val activity: SActivity) extends AnyVal with Dynamic {
         case v: Array[Double] => i.putExtra(name, v)
         case v: Array[String] => i.putExtra(name, v)
         case v: Array[CharSequence] => i.putExtra(name, v)
-        case v: Array[android.os.Parcelable] => i.putExtra(name, v)
```
Comment (guersam) [inline scaloid-common/src/main/st/org/scaloid/common/preferences.scala]: It's temporarily unqualified due to this Scalariform bug: https://github.com/daniel-trinh/scalariform/issues/44

File: scaloid-support-v4/src/main/scala/org/scaloid/support/v4/view.scala [hunk 2]
```diff
@@ -240,21 +229,16 @@ trait TraitPagerAdapter[V <: android.support.v4.view.PagerAdapter] {
...
-
 }
 
-
-
-
 trait ViewImplicits {
   import scala.language.implicitConversions
   @inline implicit def viewPager2RichViewPager[V <: android.support.v4.view.ViewPager](viewPager: V) = new RichViewPager[V](viewPager)
+  @inline implicit def pagerAdapter2RichPagerAdapter[V <: android.support.v4.view.PagerAdapter](pagerAdapter: V) = new RichPagerAdapter[V](pagerAdapter)
```
Comment (guersam) [inline scaloid-support-v4/src/main/scala/org/scaloid/support/v4/view.scala]: I don't know why it's not generated before.

**Comment by pocorall** [comment]
Great job! Thank you.

**Comment by guersam** [comment]
The future goal is to compelety replace StringTemplate with Twirl or something else, to avoid arbitrary-named 'companion templates' like `View_traitBody()`.

**Pull request merged**